Sources/Auth/AuthClient.swift

```diff
@@ -33,31 +37,39 @@ public final class AuthClient: Sendable {
   ///
   /// The session returned by this property may be expired. Use ``session`` for a session that is guaranteed to be valid.
   public var currentSession: Session? {
-    try? storage.getSession()
+    try? sessionStorage.get()
   }
 
   /// Returns the current user, if any.
   ///
   /// The user returned by this property may be outdated. Use ``user(jwt:)`` method to get an up-to-date user instance.
   public var currentUser: User? {
-    try? storage.getSession()?.user
+    try? sessionStorage.get()?.user
   }
 
   /// Namespace for accessing multi-factor authentication API.
-  public let mfa = AuthMFA()
+  public var mfa: AuthMFA {
+    AuthMFA(clientID: clientID)
+  }
+
   /// Namespace for the GoTrue admin methods.
   /// - Warning: This methods requires `service_role` key, be careful to never expose `service_role`
   /// key in the client.
-  public let admin = AuthAdmin()
+  public var admin: AuthAdmin {
+    AuthAdmin(clientID: clientID)
+  }
 
   /// Initializes a AuthClient with a specific configuration.
   ///
   /// - Parameters:
   ///   - configuration: The client configuration.
   public init(configuration: Configuration) {
-    Current = Dependencies(
+    Dependencies[clientID] = Dependencies(
```

Comment:
This should also use its own eventEmitter instead of eventEmitter.shared -- currently all auth event changes are passed to the same stream without any way of differentiating.

Reply:
You're right, thanks for bringing up the issue.

I'll PR a fix soon.

Thanks again.